### PR TITLE
1795 top3leader

### PR DESCRIPTION
**Pull request by Tthomas63**
top_3_leaderboard_

- Adds a top 3 leaderboard to the bottom of each competition detail view as a separate container.
- Styles top 3 in results with green.

### CODE REVIEW

**Comment by Tthomas63** [comment]
https://github.com/codalab/codalab-competitions/pull/1813/commits/23aad4f6b38d6ced52eccbc334de263b6cba6a33

Test failed on prior commit due to not checking length of a list. Added some more checks and resolved the issue. Also no longer displays top 3 leaderboard on competitions with no submissions.

File: codalab/apps/web/templates/web/competitions/view.html [hunk 1]
```diff
@@ -229,6 +229,29 @@ <h3 class="panel-title"><strong>Competition Ends</strong></h3>
             {% endfor %}
         </div>
     </section>
+
+    {% if topthreeleaders %}
+        <div class="phase-container" style="border: 1px solid grey;">
```
Comment (ckcollab) [inline codalab/apps/web/templates/web/competitions/view.html]: Try out the same `panel` and `panel-body` classes as on the queues page -- I think that may fit nicer with the top? 

Also, maybe we can make the top three leaderboard take up 1/3rd of the space and 2/3rds on the bottom could be the graph?

File: codalab/apps/web/models.py [hunk 2]
```diff
@@ -498,6 +498,58 @@ def get_score_headers(self):
     def get_participant_count(self):
         return self.participants.all().count()
 
+    def get_topthree(self):
```
Comment (ckcollab) [inline codalab/apps/web/models.py]: nitpick: should change this to `get_top_three`

File: codalab/apps/web/models.py [hunk 3]
```diff
@@ -498,6 +498,58 @@ def get_score_headers(self):
...
+                for index in range(num_part):
+
+                    temp_dict = {
+                        'username': local_scores[index][1]['username'],
+                        'score': local_scores[index][1]['values'][0]['val']
+                    }   
+
+                    top_threelist.append(temp_dict.copy())
+
+            return top_threelist ## Return only our top 3, with the data we want.
```
Comment (ckcollab) [inline codalab/apps/web/models.py]: nitpick: `top_threelist` -> `top_three` and for comments just do 1 `#` :)

File: codalab/apps/web/templates/web/competitions/view.html [hunk 4]
```diff
@@ -229,6 +229,29 @@ <h3 class="panel-title"><strong>Competition Ends</strong></h3>
...
+            <div class="leaderboard-result-table">
+                <table width="100%">
+                    <tr style="border: 1px solid black;">
+                        <th>Rank</th>
+                        <th>Username</th>
+                        <th>Score</th>            
+                    </tr>
+                    {% for leader in topthreeleaders %}
+                        <tr>
+                            <td>{{forloop.counter}}</td>
```
Comment (ckcollab) [inline codalab/apps/web/templates/web/competitions/view.html]: nitpicks: spaces inside brackets, like `{{ forloop.counter }}`

File: codalab/apps/web/views.py [hunk 5]
```diff
@@ -483,6 +483,14 @@ def get_context_data(self, **kwargs):
         context['site'] = Site.objects.get_current()
         context['current_server_time'] = datetime.now()
 
+        ## Top 3 Leaderboard
+
```
Comment (ckcollab) [inline codalab/apps/web/views.py]: nitpicks: remove extra lines here, single `#` instead of double

**Comment by ckcollab** [comment]
👍 on changes. All that's left, I believe:
- [ ] Pull in some charts code to be able to show a chart
- [ ] Change template a bit so Top 3 is like 1/3rd or 1/4th of the bottom, while the graph fills the remainder. Can use `<div class="col-md-4">` and `<div class="col-md-8">` probably.

File: codalab/apps/web/models.py [hunk 6]
```diff
@@ -1498,7 +1550,8 @@ def dict_constructor(loader, node):
                 comp.save()
                 logger.debug("CompetitionDefBundle::unpack saved competition logo (pk=%s)", self.pk)
             except KeyError:
-                assert False, "Could not find image {} in archive.".format(comp_base['image'])
+                assert False, "Could not find file in archive, make sure scoring_program, reference_data and logo are " \
```
Comment (Tthomas63) [inline codalab/apps/web/models.py]: Conflict here from other commit(?)

**Comment by Tthomas63** [comment]
Due to possible issues with branch, moved to a new clean branch and re-did features. 

New PR: #1823

**Pull request closed without merging**